src/operator/mshadow_op.h

```diff
@@ -842,7 +842,10 @@ struct bitwise_right_shift : public mxnet_op::tunable {
         return DType(0);
       }
     }
-    return static_cast<int64_t>(a) >> static_cast<int64_t>(b);
+    if (std::is_same<DType, uint64_t>::value)
```

Comment:
Maybe we can just return "a >> b" when is_integral<Dtype>::value is equal true
and "static_cast<int64_t>(a) >> static_cast<int64_t>(b)" otherwise.
This way we can avoid conversion of short integral representation to longer one and made this implementation more efficient.

Reply:
Good idea, it has been changed